src/desktop/apps/partner_redirect/partnerRedirectionMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+import { NextFunction } from "express"
+import { ArtsyRequest, ArtsyResponse } from "lib/middleware/artsyExpress"
+
+export const partnerRedirectionMiddleware = (
+  req: Partial<ArtsyRequest>,
+  res: Partial<ArtsyResponse>,
+  next: NextFunction
+) => {
+  if (!res.locals?.profile?.isPartner() || !res.redirect) return next()
+
+  const href: string | null = `/partner${req.path}`
+
+  // /:id/overview -> /partner/:id
+  // /:id/about -> /partner/:id
+  if (req.route.path === "/:id/overview" || req.route.path === "/:id/about") {
+    return res.redirect(301, `/partner/${req.params.id}`)
```

Comment:
I think that `req.params.id` here will be the _profile_ id, right?

If that's the case, this won't necessarily be correct. We'll have to use the partner ID/slug instead.

Reply:
We'll need to do something [similar to this](https://github.com/artsy/force/blob/master/src/desktop/apps/fair_redirect/fairRedirectionMiddleware.ts#L13).

Reply:
Updated

Reply:
Should we update the links that we updated at part 4 https://github.com/artsy/force/pull/7685 to use partner `id` instead of `default_profile_id`?

Reply:
I also think that we should update metaphysics to also use `slug` instead of `default_profile_id` at `href` field https://github.com/artsy/metaphysics/pull/3193

Reply:
@sweir27 ?